compilation fails on OSX

Commit #de91bb7

`make release-static` fails with the following error:

https://paste.fedoraproject.org/382447/64978311/

Commit e97d96ccfbb639f621cc4f4336c54cc02fa43485 breaks the tree. Right before that bitmonerod compiles fine.

`make release-static` spews the following:

`[ 62%] Building CXX object src/common/CMakeFiles/common.dir/stack_trace.cpp.o
/Users/antonis/cryptos/bitmonero-master/src/common/stack_trace.cpp:116:2: error: libunwind disabled, no stack traces [-Werror,-W#warnings]
#warning libunwind disabled, no stack traces`

while `make release` spews this one.

`[ 63%] Building CXX object src/common/CMakeFiles/common.dir/stack_trace.cpp.o
/Users/antonis/cryptos/bitmonero-master/src/common/stack_trace.cpp:44:17: error: conflicting types for '__cxa_throw'
extern "C" void CXA_THROW(void *ex, void *info, void (*dest)(void*))
                ^
/Users/antonis/cryptos/bitmonero-master/src/common/stack_trace.cpp:41:19: note: expanded from macro 'CXA_THROW'
#define CXA_THROW __cxa_throw
                  ^
/usr/include/cxxabi.h:40:32: note: previous declaration is here
extern LIBCXXABI_NORETURN void __cxa_throw(void * thrown_exception,
                               ^
/Users/antonis/cryptos/bitmonero-master/src/common/stack_trace.cpp:55:23: error: cannot initialize a variable of type 'void (*const)(void *, void *, void (*)(void *))
      __attribute__((noreturn))' with an rvalue of type 'void (*)(void *, void *, void (*)(void *))'
  static void (*const rethrow)(void*, void*, void(*)(void*)) __attribute__((noreturn)) = (void(*)(void*, void*, void(*)(void*)))dlsym(RTLD_NEXT, "__cxa_throw");
                      ^                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/antonis/cryptos/bitmonero-master/src/common/stack_trace.cpp:116:2: error: libunwind disabled, no stack traces [-Werror,-W#warnings]
#warning libunwind disabled, no stack traces
 ^
3 errors generated.
make[3]: *** [src/common/CMakeFiles/common.dir/stack_trace.cpp.o] Error 1
make[2]: *** [src/common/CMakeFiles/common.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [release] Error 2`

--- a/contrib/epee/include/net/abstract_tcp_server2.h
+++ b/contrib/epee/include/net/abstract_tcp_server2.h
@@ -95,7 +95,7 @@ namespace net_utils
 			i_connection_filter * &pfilter
 			,t_connection_type connection_type);
 
-    virtual ~connection();
+    virtual ~connection() noexcept(false);
     /// Get the socket associated with the connection.
     boost::asio::ip::tcp::socket& socket();
 

